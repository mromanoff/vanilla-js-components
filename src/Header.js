import Component from "./Component";
import inject from "./decorators/inject";

@inject("inventoryStore")
class Header extends Component {
  constructor(props, store) {
    super(props);
    this.state = { ...props }; // Spread in the props from our initial state as the state.
    this.store = store;
  }

  template() {
    // the data is passed in as props here.
    const { title } = this.state.data;
    return `
        <div class="nav">
          ${title} ${this.store.title}
        </div>
      `;
  }
}

export default Header;
