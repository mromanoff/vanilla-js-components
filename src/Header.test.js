/**
 * @jest-environment jsdom
 */

import Header from './Header';

it('Header renders correctly', () => {

  document.componentRegistry = {};
  document.nextId = 0;

  const title = 'JS Component POC';

  const element = document.createElement('div');
  element.setAttribute('data-component', 'header');
  let component = new Header({element, data: {title}});
  component.render();

  expect(component).toMatchSnapshot();
});