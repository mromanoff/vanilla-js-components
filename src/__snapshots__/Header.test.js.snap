// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Header {
  "_id": 1,
  "state": Object {
    "data": Object {
      "title": "JS Component POC",
    },
    "element": <div
      data-component="header"
    >
      
        
      <div
        class="nav"
      >
        
          JS Component POC Inventory Store title
        
      </div>
      
      
    </div>,
  },
  "store": Object {
    "a": 10,
    "b": 20,
    "title": "Inventory Store title",
  },
}
`;
